core/storage/pager.rs

```diff
@@ -5,22 +5,30 @@ use crate::storage::wal::Wal;
 use crate::{Buffer, Result};
 use log::{debug, trace};
 use sieve_cache::SieveCache;
-use std::cell::RefCell;
+use std::cell::{RefCell, UnsafeCell};
 use std::collections::{HashMap, HashSet};
 use std::hash::Hash;
 use std::ptr::{drop_in_place, NonNull};
 use std::rc::Rc;
 use std::sync::atomic::{AtomicUsize, Ordering};
-use std::sync::Arc;
+use std::sync::{Arc, RwLock};
 
 use super::wal::CheckpointStatus;
 
-pub struct Page {
+pub struct PageInner {
     pub flags: AtomicUsize,
     pub contents: Option<PageContent>,
     pub id: usize,
 }
 
+pub struct Page {
+    pub inner: UnsafeCell<PageInner>,
+}
+
+// Concurrency control of pages will be handled by the pager, we won't wrap Page with RwLock
+// because that is bad bad.
```

Comment:
for those of us with brains smooth as a new born baby's skin, can you add a little note about why this is bad bad

Reply:
I will add comments once everything is tied together. Right now this is wip